server/chat-plugins/modlog-viewer.ts

```diff
@@ -538,13 +538,17 @@ export const commands: ChatCommands = {
 		let roomid: ModlogID = (!room || room.roomid === 'staff' ? 'global' : room.roomid);
 		let lines;
 		const search: ModlogSearch = {};
-		for (const option of target.split(',')) {
+		const targets = target.split(',');
+		for (const option of targets) {
 			let [param, value] = option.split('=').map(part => part.trim());
 			if (!value) {
 				// We should guess what parameter they meant
 				value = param.trim();
 				if (/^[0-9]{1,3}\.[0-9]{1,3}/.test(value)) {
 					param = 'ip';
+				} else if (targets.indexOf(option) === 0) {
```

Comment:
```suggestion
		for (const [i, option] of targets) {
			let [param, value] = option.split('=').map(part => part.trim());
			if (!value) {
				// We should guess what parameter they meant
				value = param.trim();
				if (/^[0-9]{1,3}\.[0-9]{1,3}/.test(value)) {
					param = 'ip';
				} else if (i === 0) {
```

Reply:
Please do not ever search an array you're iterating through; there's nearly always a better way to do it.